pymc_marketing/product_incrementality/mv_its.py

```diff
@@ -0,0 +1,445 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Multivariate Interrupted Time Series Analysis for Product Incrementality."""
+
+import arviz as az
+import matplotlib.pyplot as plt
+import matplotlib.ticker as mtick
+import numpy as np
+import pandas as pd
+import pymc as pm
+
+HDI_ALPHA = 0.5
+
+
+class MVITS:
+    """Multivariate Interrupted Time Series class.
+
+    Class to perform a multivariate interrupted time series analysis with the
+    specific intent of determining where the sales of a new product came from.
+    """
+
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        treatment_time,
+        existing_sales: list[str],
+        treatment_sales: str,
+        market_saturated: bool = True,
+        rng=42,
+        sample_kwargs: dict | None = None,
+    ):
+        self.data = data
+        self.treatment_time = treatment_time
+        self.existing_sales = existing_sales
+        self.treatment_sales = treatment_sales
+        self.rng = rng
+        self.sample_kwargs = sample_kwargs if sample_kwargs is not None else {}
+        self.market_saturated = market_saturated
+
+        self.model = self.build_model(
+            self.data[self.existing_sales],
+            self.data[self.treatment_sales],
+            self.market_saturated,
+            treatment_time=self.treatment_time,
+        )
+        self.sample_prior_predictive()
+        self.fit()
+        self.sample_posterior_predictive()
+        self.calculate_counterfactual()
```

Comment:
The other classes have separation between initialization and sampling. Should be continue with that here? It also might make the testing easier. WDYT?